src/Npgsql/TypeHandling/DefaultTypeHandlerFactory.cs

```diff
@@ -28,7 +28,7 @@ internal DefaultTypeHandlerFactory(Type handlerType)
         }
 
         public NpgsqlTypeHandler Create(PostgresType pgType, NpgsqlConnection conn)
-            => (NpgsqlTypeHandler)Activator.CreateInstance(_handlerType, pgType);
+            => (NpgsqlTypeHandler)Activator.CreateInstance(_handlerType, pgType)!;
```

Comment:
Activator.CreateInstance returns nullable??

Reply:
Yes, wrote a note to ask about it later.

Reply:
We forgot about `Nullable<T>` which is a `null` object reference when boxed.